Add USB 2.0 peripheral for STM32L053 in device tree

gudnimg opened the pull request:
I noticed STM32L053 didn't have any USB FS peripheral set similar to STM32L073. It seems to be a copy-paste from STM32L073 since they are both STM32L0x3.

This pull request also:

* Removes redundant `ram-size` parameter set for STM32F469, it is already set for STM32F405 which is included into STM32F469 `.dtsi` file.
* Add `maximum-speed` parameter for USB peripheral for STM32L072 & STM32L073 and all STM32L4x2, STM32L4x3. I set it to `"full-speed"`. Saw this parameter when looking through STM32F4 `.dtsi` files.

This is my first pull request here, hope this helps. 😄 

![image](https://user-images.githubusercontent.com/8218499/109440793-4bdeb100-7a2b-11eb-9697-c81fc9f11b9a.png)

gudnimg committed:
dts: arm: st: l4: Add maximum-speed parameter to some STM32L4xx

Add optional maximum-speed parameter to USB nodes. This affects all
defined stm32l4xx, except stm32l4rx and stm32l4sx

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

erwango commented on dts/arm/st/l0/stm32l072.dtsi:
```diff
@@ -19,6 +19,7 @@
 			interrupt-names = "usb";
 			num-bidir-endpoints = <8>;
 			ram-size = <1024>;
+			maximum-speed = "full-speed";
```
This is not strictly required as this property is optional. 
When not available, USB controller set speed to the controller maximum available speed.
This is what is done in STM32 USB driver.
Anyway, this doesn't harm.

erwango reviewed (changes requested):
Thanks for this contribution.

The only change requested is to re-compose your pull request. 
-squash all the commits
-split in 2 or 3 commits: 1 commit to introduce the new node, other commit(s) for the L4/F4 fixes.
For each commit, please provide a commit message in accordance with contributing guidelines (you can also check other commit messages)

GunZi200 commented:
> Thanks for this contribution.
> 
> The only change requested is to re-compose your pull request.
> -squash all the commits
> -split in 2 or 3 commits: 1 commit to introduce the new node, other commit(s) for the L4/F4 fixes.
> For each commit, please provide a commit message in accordance with contributing guidelines (you can also check other commit messages)

Do I understand it correctly that this would require a new branch / fork to change the previous commit messages?

erwango commented:
> > Thanks for this contribution.
> > The only change requested is to re-compose your pull request.
> > -squash all the commits
> > -split in 2 or 3 commits: 1 commit to introduce the new node, other commit(s) for the L4/F4 fixes.
> > For each commit, please provide a commit message in accordance with contributing guidelines (you can also check other commit messages)
> 
> Do I understand it correctly that this would require a new branch / fork to change the previous commit messages?

You don't need a new branch, nor fork to do this. In few steps:
-1)squash all your commits into one
-2)amend the commit, spit the content, rework the commit message, commit
-3)Iterate 2) for the next commits
-4)Force push your branch (git push -f <your_fork> <your_branch_name>)

gudnimg committed:
dts: arm: st: f4: Remove redundant ram-size for stm32f469

This parameter is defined for stm32f405 which and is included into
stm32f469. The include order is as follows:

st/f4/stm32f4.dtsi
st/f4/stm32f401.dtsi
st/f4/stm32f405.dtsi <-- ram_size defined here
st/f4/stm32f407.dtsi
st/f4/stm32f427.dtsi
st/f4/stm32f429.dtsi
st/f4/stm32f469.dtsi

Refer to Application Note AN4879 from ST to view ram size for each MCU.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

gudnimg committed:
dts: arm: st: l0: Add USB FS node for stm32l053

This is the same code used to define a node for stm32l072. The two MCUs
should be the same.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

gudnimg committed:
dts: arm: st: l0: Add optional maximum-speed parameter for stm32l072

Add a maximum-speed parameter for stm32l072

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

FRASTM reviewed (commented):
LGTM including @erwango 's comments
